lib/matplotlib/pyplot.py

```diff
@@ -961,18 +961,18 @@ def subplot(*args, **kwargs):
                          "and/or 'nrows'.  Did you intend to call subplots()?")
 
     fig = gcf()
-    a = fig.add_subplot(*args, **kwargs)
-    bbox = a.bbox
+    ax = fig.add_subplot(*args, **kwargs)
+    bbox = ax.bbox
     byebye = []
     for other in fig.axes:
-        if other == a:
+        if other == ax:
             continue
         if bbox.fully_overlaps(other.bbox):
             byebye.append(other)
-    for ax in byebye:
-        delaxes(ax)
+    for ax_i in byebye:
```

Comment:
"other", as above?

Reply:
Is there a reason we have to collect the axes to delete first and then `delaxes()`in a separate loop? Seems it would be simpler to del directly in the first loop.

Reply:
Deleting an axes does a lot of stuff, not entirely sure whether it can change the positions of others, but it's just safer in two loops.